insights/tests/datasources/test_ethernet.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from insights.core.dr import SkipComponent
+from insights.specs.datasources.ethernet import interfaces, LocalSpecs
+from insights.tests import context_wrap
+
+RELATIVE_PATH = "insights_commands/ethernet_interfaces"
+
+IP_LINK = """
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000\\    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+2: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000\\    link/ether 52:54:00:13:14:b5 brd ff:ff:ff:ff:ff:ff
+3: enp8s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000\\    link/ether 52:54:00:e5:11:d4 brd ff:ff:ff:ff:ff:ff
+4: enp1s0.2@enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000\\    link/ether 52:54:00:13:14:b5 brd ff:ff:ff:ff:ff:ff
+5: ib0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 4092 qdisc mq state DOWN group default qlen 256\\    link/infiniband 00:01:02:03:fd:90:0:00:00:00:00:00:ef:0d:8b:02:01:d9:82:fd
+"""
+
+IP_LINK_BAD = ""
+
+EXPECTED = ['enp1s0', 'enp8s0', 'enp1s0.2']
+
+
+def test_ethernet_interfaces():
+    broker = {LocalSpecs.ip_link: context_wrap(IP_LINK.splitlines())}
```

Comment:
@ryan-blakley instead of using context_wrap here it would be better to use mock or something similar:
```
from mock.mock import Mock
...
ip_link_command = Mock()
ip_link_command.content = "content value"
broker = {LocalSpecs.ip_link: ip_link_command}
```